src/app/clusters/time-format-localization-server/time-format-localization-server.cpp

```diff
@@ -0,0 +1,200 @@
+using namespace chip::DeviceLayer;
+
+namespace {
+
+class TimeFormatLocalizationAttrAccess : public AttributeAccessInterface
+{
+public:
+    // Register for the Time Format Localization cluster on all endpoints.
+    TimeFormatLocalizationAttrAccess() : AttributeAccessInterface(Optional<EndpointId>::Missing(), TimeFormatLocalization::Id) {}
+
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+
+private:
+    CHIP_ERROR ReadSupportedCalendarTypes(AttributeValueEncoder & aEncoder);
+};
+
+TimeFormatLocalizationAttrAccess gAttrAccess;
+
+CHIP_ERROR TimeFormatLocalizationAttrAccess::ReadSupportedCalendarTypes(AttributeValueEncoder & aEncoder)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    DeviceLayer::AttributeList<TimeFormatLocalization::CalendarType, DeviceLayer::kMaxCalendarTypes> supportedCalendarTypes;
+
+    if (DeviceLayer::PlatformMgr().GetSupportedCalendarTypes(supportedCalendarTypes) == CHIP_NO_ERROR)
+    {
+        err = aEncoder.EncodeList([&supportedCalendarTypes](const auto & encoder) -> CHIP_ERROR {
+            for (auto type : supportedCalendarTypes)
+            {
+                ReturnErrorOnFailure(encoder.Encode(type));
+            }
+
+            return CHIP_NO_ERROR;
+        });
+    }
+    else
+    {
+        err = aEncoder.EncodeEmptyList();
+    }
+
+    return err;
+}
+
+CHIP_ERROR TimeFormatLocalizationAttrAccess::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    VerifyOrDie(aPath.mClusterId == TimeFormatLocalization::Id);
+
+    switch (aPath.mAttributeId)
+    {
+    case TimeFormatLocalization::Attributes::SupportedCalendarTypes::Id:
+        return ReadSupportedCalendarTypes(aEncoder);
+    default:
+        break;
+    }
+    return CHIP_NO_ERROR;
+}
+
+bool isSupportedCalendarTypes(uint8_t newType, uint8_t & validType)
+{
+    DeviceLayer::AttributeList<TimeFormatLocalization::CalendarType, DeviceLayer::kMaxCalendarTypes> supportedCalendarTypes;
+
+    if (DeviceLayer::PlatformMgr().GetSupportedCalendarTypes(supportedCalendarTypes) == CHIP_NO_ERROR)
+    {
+        for (auto type : supportedCalendarTypes)
+        {
+            validType = static_cast<uint8_t>(type);
+
+            if (validType == newType)
+            {
+                return true;
+            }
+        }
+    }
+    else
+    {
+        // Reset valid type if no supported calendar types found.
+        validType = 0;
+    }
+
+    return false;
+}
+
+} // anonymous namespace
+
+// =============================================================================
+// Pre-change callbacks for cluster attributes
+// =============================================================================
+
+static chip::Protocols::InteractionModel::Status
+emberAfPluginTimeFormatLocalizationOnCalendarTypeChange(chip::EndpointId EndpointId, uint8_t newType)
+{
+    chip::Protocols::InteractionModel::Status res;
```

Comment:
```suggestion
    Protocols::InteractionModel::Status res;
```
And same throughout.